core/src/main/java/org/frankframework/jdbc/JdbcFacade.java

```diff
@@ -119,30 +116,30 @@ protected DataSource getDatasource() throws JdbcException {
 			}
 
 			String dsinfo = datasource.toString();
-			log.info(getLogPrefix()+"looked up Datasource ["+dsName+"]: ["+dsinfo+"]");
+			log.info("{}looked up Datasource [{}]: [{}]", getLogPrefix(), dsName, dsinfo);
 		}
 		return datasource;
 	}
 
 	public String getDatasourceInfo() throws JdbcException {
-		String dsinfo=null;
 		if(getDatasource() instanceof TransactionalDbmsSupportAwareDataSourceProxy) {
-			// TODO let TransactionalDbmsSupportAwareDataSourceProxy.getInfo() use the same code as used here
 			return ((TransactionalDbmsSupportAwareDataSourceProxy) getDatasource()).getInfo();
 		}
-		try (Connection conn=getConnection()) {
-			DatabaseMetaData md=conn.getMetaData();
-			String product=md.getDatabaseProductName();
-			String productVersion=md.getDatabaseProductVersion();
-			String driver=md.getDriverName();
-			String driverVersion=md.getDriverVersion();
-			String url=md.getURL();
-			String user=md.getUserName();
-			dsinfo ="user ["+user+"] url ["+url+"] product ["+product+"] product version ["+productVersion+"] driver ["+driver+"] driver version ["+driverVersion+"] datasource ["+getDatasource().toString()+"]";
+		StringBuilder info = new StringBuilder();
+		try (Connection conn = getConnection()) {
+			DatabaseMetaData md = conn.getMetaData();
+			info.append("user [").append(md.getUserName()).append("], ");
+			info.append("url [").append(md.getURL()).append("], ");
+			info.append("product [").append(md.getDatabaseProductName()).append("], ");
+			info.append("product version [").append(md.getDatabaseProductVersion()).append("], ");
+			info.append("driver [").append(md.getDriverName()).append("], ");
+			info.append("driver version [").append(md.getDriverVersion()).append("], ");
+			info.append("maxConnections [").append(md.getMaxConnections()).append("], ");
+			info.append("datasource [").append(getDatasource().toString()).append("]");
```

Comment:
Since we're giving this some rework I propose throwing an IllegalStateException here. The datasource should ALWAYS be an TransactionalDataSourceProxy.

Reply:
Ah, wasn't aware of that. Will change it.